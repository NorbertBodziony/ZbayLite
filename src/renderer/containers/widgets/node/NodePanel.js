import React from 'react'
import { bindActionCreators } from 'redux'
import { connect } from 'react-redux'

import NodePanelComponent from '../../../components/widgets/node/NodePanel'
import nodeHandlers from '../../../store/handlers/node'
import { useInterval } from '../../hooks'

export const mapDispatchToProps = dispatch => bindActionCreators({
  getStatus: nodeHandlers.epics.getStatus
}, dispatch)

export const NodePanel = ({ className, getStatus }) => {
  useInterval(getStatus, 15000)
  return (
    <NodePanelComponent className={className} />
  )
}

export default connect(null, mapDispatchToProps)(NodePanel)
