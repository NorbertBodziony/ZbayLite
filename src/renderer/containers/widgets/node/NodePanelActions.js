import { bindActionCreators } from 'redux'
import { connect } from 'react-redux'

import NodePanelActions from '../../../components/widgets/node/NodePanelActions'

import nodeHandlers from '../../../store/handlers/node'

export const mapDispatchToProps = dispatch => bindActionCreators({
  onRestart: nodeHandlers.epics.restart,
  onPower: nodeHandlers.epics.togglePower
}, dispatch)

export default connect(null, mapDispatchToProps)(NodePanelActions)
