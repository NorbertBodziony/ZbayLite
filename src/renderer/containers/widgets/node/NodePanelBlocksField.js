import React from 'react'
import { connect } from 'react-redux'

import Typography from '@material-ui/core/Typography'

import NodePanelField from '../../../components/widgets/node/NodePanelField'

import nodeSelectors from '../../../store/selectors/node'

export const mapStateToProps = state => ({
  latestBlock: nodeSelectors.latestBlock(state),
  currentBlock: nodeSelectors.currentBlock(state)
})

export const NodePanelBlocksField = ({ latestBlock, currentBlock }) => (
  <NodePanelField name='Blocks'>
    <Typography inline variant='overline'>
      {currentBlock} / {latestBlock || '?'}
    </Typography>
  </NodePanelField>
)

export default connect(mapStateToProps)(NodePanelBlocksField)
