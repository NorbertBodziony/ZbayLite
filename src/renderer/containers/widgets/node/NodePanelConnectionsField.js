import React from 'react'
import BigNumber from 'bignumber.js'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'

import Typography from '@material-ui/core/Typography'

import NodePanelField from '../../../components/widgets/node/NodePanelField'

import nodeSelectors from '../../../store/selectors/node'

export const mapStateToProps = state => ({
  connections: nodeSelectors.connections(state)
})

export const NodePanelConnectionsField = ({ connections }) => (
  <NodePanelField name='Connections'>
    <Typography inline variant='overline'>
      {connections.toString()}
    </Typography>
  </NodePanelField>
)

NodePanelConnectionsField.propTypes = {
  connections: PropTypes.instanceOf(BigNumber).isRequired
}

export default connect(mapStateToProps)(NodePanelConnectionsField)
