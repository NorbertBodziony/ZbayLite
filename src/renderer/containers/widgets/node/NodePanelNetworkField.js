import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'

import Typography from '@material-ui/core/Typography'

import NodePanelField from '../../../components/widgets/node/NodePanelField'

import nodeSelectors from '../../../store/selectors/node'

export const mapStateToProps = state => ({
  network: nodeSelectors.network(state)
})

export const NodePanelNetworkField = ({ network }) => (
  <NodePanelField name='Network'>
    <Typography inline variant='overline'>
      {network || '?'}
    </Typography>
  </NodePanelField>
)

NodePanelNetworkField.propTypes = {
  network: PropTypes.oneOf(['mainnet', 'testnet'])
}

export default connect(mapStateToProps)(NodePanelNetworkField)
