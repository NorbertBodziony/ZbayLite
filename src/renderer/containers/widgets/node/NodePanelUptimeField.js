import React from 'react'
import PropTypes from 'prop-types'
import * as R from 'ramda'
import { connect } from 'react-redux'

import Typography from '@material-ui/core/Typography'

import NodePanelField from '../../../components/widgets/node/NodePanelField'

import nodeSelectors from '../../../store/selectors/node'

export const mapStateToProps = state => ({
  uptime: nodeSelectors.uptime(state)
})

export const NodePanelUptimeField = ({ uptime }) => (
  <NodePanelField name='Uptime'>
    <Typography display='inline' variant='overline'>
      {R.isEmpty(uptime) ? '-' : null}
      {uptime.days ? `${uptime.days}d  ` : null}
      {uptime.hours ? `${uptime.hours}h  ` : null}
      {uptime.minutes ? `${uptime.minutes}m` : null}
    </Typography>
  </NodePanelField>
)

NodePanelUptimeField.propTypes = {
  uptime: PropTypes.exact({
    days: PropTypes.number,
    hours: PropTypes.number,
    minutes: PropTypes.number,
    seconds: PropTypes.number
  }).isRequired
}

export default connect(mapStateToProps)(NodePanelUptimeField)
