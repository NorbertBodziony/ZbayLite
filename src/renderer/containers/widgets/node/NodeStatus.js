import { connect } from 'react-redux'

import NodeStatus from '../../../components/widgets/node/NodeStatus'

import nodeSelectors from '../../../store/selectors/node'
import identitySelectors from '../../../store/selectors/identity'

export const mapStateToProps = state => ({
  status: nodeSelectors.status(state),
  percentSynced: nodeSelectors.percentSynced(state),
  freeUtxos: identitySelectors.freeUtxos(state)
})

export default connect(mapStateToProps)(NodeStatus)
