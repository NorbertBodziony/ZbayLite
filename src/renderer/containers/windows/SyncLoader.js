import React, { useEffect } from 'react'
import { Redirect } from 'react-router'
import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'

import SyncLoaderComponent from '../../components/windows/SyncLoader'
import nodeSelectors from '../../store/selectors/node'
import identitySelectors from '../../store/selectors/identity'
import nodeHandlers from '../../store/handlers/node'
import appHandlers from '../../store/handlers/app'
import vaultHandlers from '../../store/handlers/vault'
import vaultSelectors from '../../store/selectors/vault'
import { actionCreators } from '../../store/handlers/modals'
import electronStore from '../../../shared/electronStore'

import { useInterval } from '../hooks'

export const mapStateToProps = state => ({
  node: nodeSelectors.node(state),
  creating: vaultSelectors.creating(state),
  exists: vaultSelectors.exists(state),
  locked: vaultSelectors.locked(state),
  bootstrapping: nodeSelectors.bootstrapping(state),
  bootstrappingMessage: nodeSelectors.bootstrappingMessage(state),
  nodeConnected: nodeSelectors.isConnected(state),
  fetchingStatus: nodeSelectors.fetchingStatus(state),
  fetchingPart: nodeSelectors.fetchingPart(state),
  fetchingSizeLeft: nodeSelectors.fetchingSize(state),
  fetchingSpeed: nodeSelectors.fetchingSpeed(state),
  fetchingEndTime: nodeSelectors.fetchingEndTime(state),
  hasAddress: identitySelectors.address(state),
  isRescanningMonitorStarted: nodeSelectors.isRescanningMonitorStarted(state),
  rescanningProgress: nodeSelectors.rescanningProgress(state),
  isFetching: nodeSelectors.isFetching(state)
})
export const mapDispatchToProps = dispatch =>
  bindActionCreators(
    {
      getStatus: nodeHandlers.epics.getStatus,
      openModal: actionCreators.openModal('topUp'),
      setVaultIdentity: vaultHandlers.epics.setVaultIdentity,
      skip: appHandlers.actions.setNewUser,
      startRescanningMonitor: nodeHandlers.epics.startRescanningMonitor,
      disablePowerSaveMode: nodeHandlers.epics.disablePowerSaveMode
    },
    dispatch
  )

export const SyncLoader = ({ setVaultIdentity, isFetching, disablePowerSaveMode, isRescanningMonitorStarted, rescanningProgress, startRescanningMonitor, hasAddress, node, getStatus, bootstrapping, bootstrappingMessage, nodeConnected, openModal, creating, locked, exists, fetchingPart, fetchingSizeLeft, fetchingStatus, fetchingEndTime, fetchingSpeed }) => {
  const blockchainStatus = electronStore.get('AppStatus.blockchain.status')
  const isBlockchainRescanned = electronStore.get('AppStatus.blockchain.isRescanned')
  const lastBlock = node.latestBlock.isEqualTo(0) ? 999999 : node.latestBlock
  const sync = parseFloat(node.currentBlock.div(lastBlock) * 100).toFixed(2)
  const fetching = ((((26202539059 - (fetchingSizeLeft)) * 100) / 26202539059)).toFixed()
  let ETA = null
  if (fetchingEndTime) {
    const { hours, minutes } = fetchingEndTime
    ETA = `${hours ? `${hours}h` : ''} ${minutes ? `${minutes}m left` : ''}`
  }
  const rescanningProgressInPercents = rescanningProgress / 722000 * 100
  const isFetchingComplete = ((fetchingPart === 'blockchain' && fetchingStatus === 'SUCCESS') || blockchainStatus === 'SUCCESS')
  const isRescaningComplete = ((isBlockchainRescanned || rescanningProgressInPercents > 80) && !node.latestBlock.isEqualTo(0))
  const rescanningWithFetchingPartProgress = 50 + (rescanningProgressInPercents / 2)
  let progressValue
  let message
  useInterval(getStatus, 15000)
  useEffect(
    () => {
      if ((!locked && nodeConnected && fetchingPart === 'blockchain' && fetchingStatus === 'SUCCESS') || blockchainStatus === 'SUCCESS') {
        if (nodeConnected) {
          setVaultIdentity()
        }
        if (!isRescanningMonitorStarted) {
          startRescanningMonitor()
        }
      }
    },
    [nodeConnected, fetchingStatus, fetchingPart, locked]
  )
  useEffect(
    () => {
      if (isFetchingComplete && isRescaningComplete) {
        setVaultIdentity()
        startRescanningMonitor()
        disablePowerSaveMode()
      }
    },
    [isFetchingComplete, isRescaningComplete]
  )
  if (!isFetchingComplete) {
    const fetchingProgress = fetching === '100' ? 0 : fetching / 2
    progressValue = fetchingProgress
  } else if (!isRescaningComplete) {
    progressValue = rescanningWithFetchingPartProgress
    message = `Rescanning blocks, ${rescanningProgress ? ETA || '~40 minutes left' : '~40 minutes left'} (${rescanningProgress}/722000)`
  } else {
    electronStore.set('AppStatus.blockchain.isRescanned', true)
    progressValue = sync
    message = `Final sync (${node.currentBlock}/${lastBlock})`
  }
  return (locked || (node.latestBlock.lt(400000) || node.latestBlock.minus(node.currentBlock).gt(10))) ? (
    <SyncLoaderComponent
      isBlockchainRescanned={isBlockchainRescanned}
      isRescanningMonitorStarted={isRescanningMonitorStarted}
      rescanningProgress={rescanningProgress}
      fetchingEndTime={fetchingEndTime}
      fetchingSpeed={fetchingSpeed}
      hasAddress={hasAddress}
      node={node}
      blockchainStatus={blockchainStatus}
      bootstrapping={bootstrapping}
      bootstrappingMessage={bootstrappingMessage}
      openModal={openModal}
      nodeConnected={nodeConnected}
      fetchingPart={fetchingPart}
      fetchingSizeLeft={fetchingSizeLeft}
      fetchingStatus={fetchingStatus}
      ETA={ETA}
      progressValue={progressValue}
      isFetching={isFetching}
      message={message} />
  ) : (
    <Redirect to='/main/channel/general' />
  )
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(SyncLoader)
