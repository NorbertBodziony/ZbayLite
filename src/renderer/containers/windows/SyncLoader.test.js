import Immutable from 'immutable'

import { mapStateToProps } from './SyncLoader'
import create from '../../store/create'
import { NodeState } from '../../store/handlers/node'

describe('SyncLoader', () => {
  let store = null
  beforeEach(() => {
    jest.clearAllMocks()
    store = create({
      initialState: Immutable.Map({
        node: NodeState({
          currentBlock: 1,
          latestBlock: 100
        })
      })
    })
  })

  it('will receive right props', async () => {
    const props = mapStateToProps(store.getState())
    expect(props).toMatchSnapshot()
  })
})
