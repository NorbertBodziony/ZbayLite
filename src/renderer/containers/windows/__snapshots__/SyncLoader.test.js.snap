// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SyncLoader will receive right props 1`] = `
Object {
  "node": Immutable.NodeState {
    "latestBlock": 100,
    "currentBlock": 1,
    "connections": "0",
    "isTestnet": null,
    "status": "connecting",
    "errors": "",
    "bootstrapLoader": Immutable.Record {
      "loading": false,
      "message": "",
    },
    "startedAt": null,
  },
}
`;
